src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,56 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Class<?>> interfaceImplementations;
+
+    private Injector() {
+        interfaceImplementations = Map.of(
+                FileReaderService.class, FileReaderServiceImpl.class,
+                ProductParser.class, ProductParserImpl.class,
+                ProductService.class, ProductServiceImpl.class
+        );
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implClass = interfaceImplementations.get(interfaceClazz);
+
+        if (implClass == null || !implClass.isAnnotationPresent(Component.class)) {
```

Comment:
Before throwing an exception due to a missing `@Component` annotation, you should check if `implClass` is not null to avoid a NullPointerException. The current condition will fail if `implClass` is null because the second part of the condition will be evaluated and result in a NullPointerException.